lib/chef/provider/cron.rb

```diff
@@ -206,13 +206,15 @@ def write_crontab(crontab)
         write_exception = false
 
         tempname = Dir::Tmpname.create(["crontab-"]) {}
-        TargetIO::File.open(tempname, "w") do |tempfile|
+        io = TargetIO::File.open(tempname, "w") do |tempfile|
           tempfile.write(crontab)
         end
 
+        tempname = io.path if ChefConfig::Config.target_mode?
+
         so = shell_out!("crontab -u #{new_resource.user} #{tempname}")
 
-        TargetIO::File.unlink(tempname)
+        TargetIO::File.unlink(tempname) if ChefConfig::Config.target_mode?
```

Comment:
we still need to remove the local tmpfile, even if we're not in targetmode, just like we did before.

Reply:
Ouch! Good catch, thanks.

Reply:
@thheinen is this updated per Phil's last comment, or can we merge it?